File: WebUI/script/types/AABB.js
class AABB {
    constructor(min, max, transform) {
        this.min = min;
        this.max = max;
        this.transform = transform;
    }

    setFromTable(table) {
        this.min = new Vec3().setFromString(table.min);
        this.max = new Vec3().setFromString(table.max);
        this.transform = new LinearTransform().setFromTable(table.transform);

        return this;
    }
}

File: ext/Shared/Types/AABB.lua
class 'AABB'

local m_Logger = Logger("AABB", true)

function AABB:__init(p_Arg)
	self.min = p_Arg.min
	self.max = p_Arg.max
	self.transform = p_Arg.transform
end

function AABB:GetTable()
	return {
		min = self.min,
		max = self.max,
		transform = self.transform
	}
end

return AABB
